File: structy/array_and_strings/2_compress.js
const compress = (s) => {
  // todo
  let count = 1;
  let pointer1 = 0;
  let pointer2 = 1;
  let result = "";
  
  while(pointer2 <= s.length){
    if(s[pointer1] === s[pointer2]){
      count++;
    }else{
      if(count === 1){
        result = result.concat(s[pointer1]);
      }else{
        result = result.concat(count.toString(), s[pointer1]);
      }
      count = 1;
    }
    pointer1++;
    pointer2++;
  }

  return result;
};

// n = length of string
// Time: O(n)
// Space: O(n)

File: structy/Python/2_arrays_and_strings/2_compress.py
def compress(s):
  res = ''
  count = 0
  curr_char = s[0]
  
  for char in s:
    if curr_char == char:
      count = count + 1
    else:
      if count == 1:
        res += curr_char
      else:
        res += str(count) + curr_char
        
      curr_char = char
      count = 1
        
  if count == 1: 
    res += curr_char
  else:
    res += str(count) + curr_char 
    
  return res

  pass # todo